### ISSUE TITLE
scANVI prediction with controlled labels size

### ISSUE THREAD

**Issue by michalmark** [YosefLab/CZI-Immuneaging#86]
Hi Can, 

The scANVI prediction on the T cells using the NY classifier as cell type labels,  gives not very good predicted labels.  
The adata 5hd file and the scANVI model are in  the: "/data/yosef2/users/can/exchange_folder/valeh/scanvi_NY_classification/" 

You suggested to use this function, what is the 'subsampled_labels' refers to?: 
yprior = torch.tensor([
        adata.obs['classification'].value_counts()[i]/adata.n_obs for i in
        adata.obs['subsampled_labels'].cat.categories
         if i is not adata.uns['Unknown']])

The code I used is here: 
```
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import scanpy as sc
import scvi
import anndata as ad
import torch

adata=sc.read("/data/yosef2/users/can/exchange_folder/valeh/scanvi_NY_classification/adata_clean_NY.h5ad")

scvi.data.setup_anndata(adata, batch_key='donor_id', labels_key = "classification")
scanvi = scvi.model.SCANVI(adata, unlabeled_category="Unknown")
scanvi.train(n_samples_per_label=100)

```
To upload the data directly 
```
model_SCANVI_path= "/data/yosef2/users/can/exchange_folder/valeh/scanvi_NY_classification/"
scanvi = scvi.model.SCANVI.load(model_SCANVI_path, adata)
adata.obs["Predicted_scANVI"] = scanvi.predict(adata)
adata.obsm["X_scANVI"] = scanvi.get_latent_representation(adata)

df = adata.obs.groupby(["classification", "Predicted_scANVI"]).size().unstack(fill_value=0)
import matplotlib.pyplot as plt
df = df / df.sum(axis=0)

plt.figure(figsize=(8, 8))
_ = plt.pcolor(df)
_ = plt.xticks(np.arange(0.5, len(df.columns), 1), df.columns, rotation=90)
_ = plt.yticks(np.arange(0.5, len(df.index), 1), df.index)
plt.xlabel("Predicted")
plt.ylabel("Observed")

 
```
 
![prediction](https://user-images.githubusercontent.com/103484145/216847114-5be7ce4b-068f-416d-9774-125e1fad7b78.png)
